example/lib/map_recorder_example.dart

```diff
@@ -0,0 +1,271 @@
+class MapRecorderExample extends StatefulWidget implements Example {
+  @override
+  final Widget leading = const Icon(Icons.fiber_smart_record);
+  @override
+  final String title = 'Map Recorder';
+  @override
+  final String? subtitle = 'Record and replay map sessions';
+
+  @override
+  State<StatefulWidget> createState() => MapRecorderExampleState();
+}
+
+class MapRecorderExampleState extends State<MapRecorderExample> {
+  MapboxMap? mapboxMap;
+  Uint8List? recordedSequence;
+  bool isRecording = false;
+  bool isReplaying = false;
+  String playbackState = 'IDLE';
+
+  _onMapCreated(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+  }
+
+  Future<void> _startRecording() async {
+    if (mapboxMap == null) return;
+
+    setState(() {
+      isRecording = true;
+      recordedSequence = null;
+    });
+
+    try {
+      await mapboxMap!.recorder.startRecording(
+        timeWindow: 60000, // Keep last 60 seconds
+        loggingEnabled: true,
+        compressed: true,
+      );
+
+      ScaffoldMessenger.of(context).showSnackBar(
+        const SnackBar(
+          content: Text('Recording started. Interact with the map!'),
+          duration: Duration(seconds: 2),
+        ),
+      );
+
+      // Perform some camera animations to record
+      await Future.delayed(const Duration(milliseconds: 500));
+      await mapboxMap!.flyTo(
+        CameraOptions(
+          center: Point(coordinates: Position(-73.581, 45.4588)),
+          zoom: 11.0,
+          pitch: 35.0,
+          bearing: 90.0,
+        ),
+        MapAnimationOptions(duration: 5000, startDelay: 0),
+      );
+    } catch (e) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(content: Text('Error starting recording: $e')),
+      );
+      setState(() {
+        isRecording = false;
+      });
+    }
+  }
+
+  Future<void> _stopRecording() async {
+    if (mapboxMap == null) return;
+
+    try {
+      final sequence = await mapboxMap!.recorder.stopRecording();
+      setState(() {
+        isRecording = false;
+        recordedSequence = sequence;
+      });
+
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text('Recording stopped. ${sequence.length} bytes recorded.'),
+          duration: const Duration(seconds: 2),
+        ),
+      );
+    } catch (e) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(content: Text('Error stopping recording: $e')),
+      );
+      setState(() {
+        isRecording = false;
+      });
+    }
+  }
+
+  Future<void> _replayRecording() async {
+    if (mapboxMap == null || recordedSequence == null) return;
+
+    setState(() {
+      isReplaying = true;
+    });
+
+    try {
+      // Replay twice at 2x speed
+      await mapboxMap!.recorder.replay(
+        recordedSequence!,
+        playbackCount: 2,
+        playbackSpeedMultiplier: 2.0,
+        avoidPlaybackPauses: false,
+      );
+
+      setState(() {
```

Comment:
Not a critical issue as this is just an example, but it is worth noting that the `replay()` method is async. If the user navigates away during replay, the widget could be disposed, and calling `setState()` on a disposed widget will throw an exception.

This could be fixed by wrapping in an `if (mounted)` block like this: 

```dart
if (mounted) {
  setState(() {
    isReplaying = false;
  });
}
```

The same applies to the other setState and error calls here.